src/com/google/javascript/jscomp/CodeGenerator.java

```diff
@@ -1085,7 +1085,7 @@ protected void add(Node n, Context context) {
             checkState(
                 !isInClass, "initializers should only exist in object literals, not classes");
             cc.addOp(":", false);
-            add(initializer);
+            addExpr(initializer, NodeUtil.precedence(Token.COMMA) + 1, Context.OTHER);
```

Comment:
This should be consistent with the behavior for a normal, STRING_KEY, type of property.
See method `addStringKey()`
tl;dr is to make this
`addExpr(initializer, 1, Context.OTHER)`